message_ix/tests/test_cli.py

```diff
@@ -57,20 +55,30 @@ def test_copy_model(monkeypatch, message_ix_cli, tmp_path, tmp_env):
         "--branch=main",
         "--tag=v1.2.0",
         # Nonexistent tag
-        pytest.param("--tag=v999", marks=pytest.mark.xfail(raises=AssertionError)),
+        pytest.param("--tag=v999", marks=pytest.mark.xfail(raises=ValueError)),
+        # Can't use both --tag and --branch
+        pytest.param(
+            ["--branch=main", "--tag=v1.2.0"],
+            marks=pytest.mark.xfail(raises=click.UsageError),
+        ),
     ],
 )
 def test_dl(message_ix_cli, opts, tmp_path):
-    r = message_ix_cli("dl", opts, str(tmp_path))
-
-    assert r.exit_code == 0, (r.exception, r.output)
-
-    if opts == "":
-        # Guess what the latest release will be from GitHub, using __version__.
-        # This string is provided by setuptools-scm based on the most recent
-        # Git tag, e.g. if the tag is 'v3.0.0' it may be '3.0.1.devN+etc'.
-        major = message_ix.__version__.split(".")[0]
-
-        # 'message-ix dl' defaults to the latest release
-        pattern = re.compile(rf"Default: latest release v{major}\.\d+\.\d+")
-        assert pattern.match(r.output)
+    if isinstance(opts, list):
```

Comment:
In order to avoid this branch & duplication, use a list of (possibly 0- or 1-element) lists in the `parametrize()` call:
```python
@pytest.mark.parametrize(
    "opts",
    [
        [],
        ["--branch=main"],
        ["--tag=v1.2.0"],
        # Nonexistent tag
        pytest.param(["--tag=v999"], marks=pytest.mark.xfail(raises=AssertionError)),
        pytest.param(["--tag=v999"], marks=pytest.mark.xfail(raises=ValueError)),
        # Can't use both --tag and --branch
        pytest.param(
            ["--branch=main", "--tag=v1.2.0"],
            marks=pytest.mark.xfail(raises=click.UsageError),
        ),
    ],
)
```

Reply:
Yes, I thought about that, too. This would probably be nicer if the last case really raised a `click.UsageError`, but I'll include this in a commit, anyway.